components/ipfs/ipfs_service.cc

```diff
@@ -320,23 +325,18 @@ bool IpfsService::IsDaemonLaunched() const {
 }
 
 void IpfsService::LaunchDaemon(LaunchDaemonCallback callback) {
-  // Reject if previous launch request in progress.
-  if (launch_daemon_callback_) {
-    std::move(callback).Run(false);
-    return;
-  }
-
-  if (allow_ipfs_launch_for_test_) {
+  if (IsDaemonLaunched()) {
     std::move(callback).Run(true);
     return;
   }
 
-  if (IsDaemonLaunched()) {
-    std::move(callback).Run(true);
+  // Wait if previous launch request in progress.
+  if (!pending_launch_callbacks_.empty()) {
+    pending_launch_callbacks_.push(std::move(callback));
```

Comment:
This handling is not quite right because if you load 1 IPFS URI:
`ipfs://bafybeifk6th5qhox7pffjqjerbjxkpmsmufdcswdgacnmyv3fn53z2wgwe/`
Then open another tab and try to load another (even the same):
`ipfs://bafybeifk6th5qhox7pffjqjerbjxkpmsmufdcswdgacnmyv3fn53z2wgwe/`

The second one will spin forever. Because it doesn't know the Daemon is already started.

Reply:
in this case it will be returned(and callback called) earlier on line 330, because ipfs daemon started, also there is additional check on daemon launch inside throttler that will skip daemon launch call and move to fetching connected peers, added new test about it

Reply:
ok 👍